WalletWasabi/WabiSabi/Models/DecompositionAlgs/GreedyDecomposer.cs

```diff
@@ -0,0 +1,67 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class GreedyDecomposer
+	{
+		public GreedyDecomposer(IEnumerable<Money> preferredDenominations, Money dustThreshold, FeeRate feeRate)
+		{
+			Decomposition = new(preferredDenominations);
+			DustThreshold = dustThreshold;
+			FeeRate = feeRate;
+		}
+
+		public Decomposition Decomposition { get; }
+		private Money DustThreshold { get; }
+		private FeeRate FeeRate { get; }
+
+		public void Decompose(Coin coin)
+		{
+			Money remaining = coin.Amount - FeeRate.GetFee(coin.ScriptPubKey.EstimateOutputVsize());
+
+			while (remaining > DustThreshold)
+			{
+				Money? denom = LargestDenomBelowIncl(remaining);
+				if (denom is null || denom == Money.Zero)
+				{
+					break;
```

Comment:
Here you could add the following and remove the `if` statement at the end of the method.
```suggestion
					Decomposition.AddSorted(remaining);
					break;
```

Reply:
Why? While can finito if `remaining > DustThreshold` Still the remaining part will only be added if it is above the DustThreshold. Otherwise, it is just swallowed.

Reply:
yess!!